modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java

```diff
@@ -243,7 +245,8 @@ public ObjectField addOrUpdateSystemObjectField(
 	public ObjectField addSystemObjectField(
 			long userId, long objectDefinitionId, String businessType,
 			String dbColumnName, String dbTableName, String dbType,
-			boolean indexed, boolean indexedAsKeyword, String indexedLanguageId,
+			String externalReferenceCode, boolean indexed,
+			boolean indexedAsKeyword, String indexedLanguageId,
```

Comment:
@seltong always follow the same order from the [service.xml](https://github.com/liferay/liferay-portal/blob/b1e26f4397111b36aa8e3778179cc11631584b52/modules/apps/object/object-service/service.xml#L225-L234), so the `externalReferenceCode` must come before the `userId`